rust-query-crlite/src/main.rs

```diff
@@ -0,0 +1,794 @@
+        false => Ok(CmdResult::NoneRevoked),
+    }
+}
+
+/// A standalone tool for querying a Firefox CRLite database.
+///
+/// Subcommands exit with code 0 if none of the queried certificates are known to be revoked and
+/// with code 1 otherwise. An exit code of 0 might mean that the certificate's issuer is not
+/// enrolled or that the certificate is not covered by the filter.  Additional information is
+/// logged to stderr.
+#[derive(Parser)]
+struct Cli {
+    /// Download a new CRLite filter and associated metadata from Firefox Remote Settings.
+    #[clap(long, arg_enum)]
+    update: Option<RemoteSettingsInstance>,
+
+    /// CRLite directory e.g. <firefox profile>/security_state/.
+    #[clap(short, long, parse(from_os_str), default_value = "./crlite_db/")]
+    db: PathBuf,
+
+    /// Verbosity. -v => warning, -vv => info, -vvv => debug.
+    #[clap(short = 'v', parse(from_occurrences))]
+    verbose: usize,
+
+    #[clap(subcommand)]
+    command: Subcommand,
+}
+
+#[derive(Clone, clap::ArgEnum)]
+enum RemoteSettingsInstance {
+    Prod,
+    Stage,
+    Dev,
+}
+
+#[derive(Parser, Debug)]
+enum Subcommand {
+    /// Query certificates from TLS handshakes to one or more hosts.
+    Https { hosts: Vec<String> },
+    /// Fetch a list of https hosts from `host_file_url`, obtain a certificate from each host, and
+    /// query each certificate.
+    Signoff { host_file_url: String },
+    /// Query DER or PEM encoded certificates from one or more files.
+    X509 { files: Vec<PathBuf> },
+}
+
+enum CmdResult {
+    SomeRevoked,
+    NoneRevoked,
+}
+
+fn main() {
+    let args = Cli::parse();
+
+    stderrlog::new()
+        .module(module_path!())
+        .verbosity(args.verbose)
+        .init()
+        .unwrap();
+
+    if std::fs::create_dir_all(&args.db).is_err() {
+        error!("Could not create directory: {}", args.db.display());
+        std::process::exit(1);
+    }
+
+    if args.update.is_some() {
+        let (attachment_url, collection_url) = match args.update.unwrap() {
+            RemoteSettingsInstance::Dev => (STAGE_ATTACH_URL, STAGE_URL),
+            RemoteSettingsInstance::Stage => (STAGE_ATTACH_URL, STAGE_URL),
+            RemoteSettingsInstance::Prod => (PROD_ATTACH_URL, PROD_URL),
+        };
+
+        if let Err(e) = update_db(&args.db, attachment_url, collection_url) {
+            error!("{}", e.message);
+            std::process::exit(1);
+        }
+    }
+
+    if !(args.db.join("crlite.filter").exists()
+        && args.db.join("crlite.stash").exists()
+        && args.db.join("crlite.enrollment").exists()
+        && args.db.join("crlite.coverage").exists())
+    {
+        error!("CRLite DB is incomplete. Use --update [prod | stage] to populate");
+        std::process::exit(1);
+    }
+
+    let db = CRLiteDB::load(&args.db);
+    if let Err(e) = db {
+        error!("Error loading CRLite DB: {}", e.message);
+        std::process::exit(1);
+    }
+    let db = db.unwrap();
```

Comment:
I thiiiink you can do a match again to avoid the unwrap, but I don't know if the compiler is smart enough to know that the exit doesn't continue code flow.